tests/pipelines/earliest.nf.test

```diff
@@ -0,0 +1,235 @@
+            assert results.text.contains("sample16,one_char_one_date,a,,,,,,,2000-01-01,,False,At least one of the date values are incorrectly formatted.")
+            assert results.text.contains("sample17,many_integer_one_date,1,2,3,4,5,6,7,2000-01-01,,False,At least one of the date values are incorrectly formatted.")
+            assert results.text.contains("sample18,many_float_one_date,0.1,0.2,0.3,0.4,0.5,0.6,0.7,2000-01-01,,False,At least one of the date values are incorrectly formatted.")
+            assert results.text.contains("sample19,many_string_one_date,cat,dog,fish,hamster,rat,mouse,bird,2000-01-01,,False,At least one of the date values are incorrectly formatted.")
+            assert results.text.contains("sample20,many_char_one_date,a,b,c,d,e,f,g,2000-01-01,,False,At least one of the date values are incorrectly formatted.")
+            assert results.text.contains("sample21,one_integer_many_date,1,2000-01-01,2000-01-02,2000-01-03,2000-01-04,2000-01-05,2000-01-06,2000-01-07,,False,At least one of the date values are incorrectly formatted.")
+            assert results.text.contains("sample22,one_float_many_date,0.1,2000-01-01,2000-01-02,2000-01-03,2000-01-04,2000-01-05,2000-01-06,2000-01-07,,False,At least one of the date values are incorrectly formatted.")
+            assert results.text.contains("sample23,one_string_many_date,cat,2000-01-01,2000-01-02,2000-01-03,2000-01-04,2000-01-05,2000-01-06,2000-01-07,,False,At least one of the date values are incorrectly formatted.")
+            assert results.text.contains("sample24,one_char_many_date,a,2000-01-01,2000-01-02,2000-01-03,2000-01-04,2000-01-05,2000-01-06,2000-01-07,,False,At least one of the date values are incorrectly formatted.")
+            assert results.text.contains("sample25,duplicates,2000-01-01,2000-01-01,2000-01-01,2000-01-01,2000-01-01,2000-01-01,2000-01-01,2000-01-01,2000-01-01,True,")
+            assert results.text.contains("sample26,duplicates_missing,2000-01-01,,2000-01-01,,2000-01-01,,2000-01-01,,2000-01-01,True,")
+            assert results.text.contains("sample27,duplicates_mixed,2000-01-01,2000-01-01,2000-01-01,2000-01-01,2000-01-02,2000-01-02,2000-01-02,2000-01-02,2000-01-01,True,")
+
+            // Check IRIDA Next JSON Output
+            def iridanext_json = path("$launchDir/results/iridanext.output.json").json
+            def iridanext_global = iridanext_json.files.global
+            def iridanext_metadata = iridanext_json.metadata.samples
+
+            assert iridanext_global.findAll { it.path == "transformation/results.csv" }.size() == 1
+
+            assert iridanext_metadata.size() == 27
+
+            assert iridanext_metadata.containsKey("sample1")
+            assert iridanext_metadata.sample1.earliest_date == "2000-01-01"
+
+            assert iridanext_metadata.containsKey("sample2")
+            assert iridanext_metadata.sample2.earliest_date == ""
```

Comment:
I think it makes sense to delete earliest_date metadata here (as is happing when passing an empty string as the value), so that people don't get mixed up with any previous earliest date results.

That is to say, what you are doing right now is perfect.

Reply:
The other thing too is that so long as at least one sample has a valid earliest date, the `nf-iridanext` plugin will parse the empty string in the CSV file and write `""`. As I understand, there's no way to tell the plugin to not write empty values, unless we change the plugin code.

Reply:
Yes, that's correct. We'd have to do some additional work if we wanted to avoid deleting these values from IRIDA Next. But, how it is right now I think is the proper way.